apiserver/facades/client/usermanager/usermanager.go

```diff
@@ -596,3 +596,8 @@ func (api *UserManagerAPI) getUserByTag(ctx context.Context, tag string) (coreus
 	}
 	return api.userService.GetUserByName(ctx, userTag.Name())
 }
+
+// getUserTag returns a user tag. It is a helper function to get a user tag.
+func (api *UserManagerAPI) getUserTag(name string) names.UserTag {
```

Comment:
Have a method just to call `NewLocalUserTag` doesn't justify the overhead. 

In reset password logic, just do `api.apiUser.Id()  != usr.Name`
In the other places, just use `names.NewLocalUserTag()` directly.

But we don't even need it anyway. We already have the user tag. In fact, there is a mistake because we are unnecessarily parsing the user tag twice.

`userTag, err := names.ParseUserTag(arg.Tag)`

`usrTag, err := names.ParseUserTag(arg.Tag)`

And even after parsing it twice, we look it up again a third time ` api.getUserTag(usr.Name)`

Why are we calculating the user tag so many times?

Reply:
thx; I provided a fix that optimizes the number of user tag calculations.